docs/developer-guide/extensions/proxy-extensions.md

```diff
@@ -111,6 +114,34 @@ Defines a list with backend url by cluster.
 
 Is the address where the extension backend must be available.
 
+#### `extensions.backend.services.headers` (*list*)
+
+If provided, the headers list will be added on all outgoing requests
+for this service config. Existing headers in the incoming request with
+the same name will be overriden by the one in this list. Reserved header
+names will be ignored (see the [headers](#incoming-request-headers) below).
+
+#### `extensions.backend.services.headers.name` (*string*)
+(mandatory)
+
+Defines the name of the header. It is a mandatory field if a header is
+provided.
+
+#### `extensions.backend.services.headers.value` (*string*)
+(mandatory)
+
+Defines the value of the header. It is a mandatory field if a header is
+provided. The value can be provided as verbatim or as a reference to an
+Argo CD secret key. In order to provide it as a reference, it is
+necessary to prefix it with a dollar sign.
+
+Example:
+
+    value: '$some.argocd.secret.key'
+
+In the example above, the value will be replaced with the one from
+the argocd-secret with key 'some.argocd.secret.key'.
```

Comment:
```suggestion
In the example above, the value will be replaced with the one from
the argocd-secret with key 'some.argocd.secret.key'.

Unlike some other fields in argocd-cm, you cannot currently specify a
secret other than argocd-secret as the source for this value.
```

Is that accurate?

Reply:
You can.. it is just a matter of adding the label `app.kubernetes.io/part-of=argocd` in the secret.

Reply:
Oh dope, forget this then.